www-apps/otrs/otrs-6.0.6.ebuild

```diff
@@ -0,0 +1,156 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+inherit user systemd
+
+DESCRIPTION="OTRS is an Open source Ticket Request System"
+HOMEPAGE="https://www.otrs.com/"
+SRC_URI="https://ftp.otrs.org/pub/${PN}/${P}.tar.bz2"
+
+LICENSE="AGPL-3+"
+KEYWORDS="~amd64 ~x86"
+IUSE="apache2 fastcgi +gd ldap mod_perl +mysql pdf postgres soap"
+SLOT="0"
+
+REQUIRED_USE="|| ( mysql postgres )"
+
+DEPEND="media-libs/libpng:0"
+
+RDEPEND="dev-perl/Apache-Reload
+	dev-perl/Archive-Zip
+	dev-perl/DBI
+	dev-perl/IO-Socket-SSL
+	dev-perl/JSON-XS
+	dev-perl/LWP-UserAgent-Determined
+	dev-perl/Mail-POP3Client
+	dev-perl/Mail-IMAPClient
+	>dev-perl/Net-DNS-0.60
+	dev-perl/Template-Toolkit
+	dev-perl/Text-CSV_XS
+	dev-perl/TimeDate
+	dev-perl/DateTime
+	dev-perl/XML-LibXML-Simple
+	dev-perl/XML-Parser
+	dev-perl/YAML-LibYAML
+	apache2? (
+		mod_perl? (
+			www-servers/apache:2
+			=www-apache/libapreq2-2* www-apache/mod_perl
+		)
+		!fastcgi? ( !mod_perl? ( www-servers/apache:2[suexec] ) )
+	)
+	fastcgi? (
+		dev-perl/FCGI
+		virtual/httpd-fastcgi
+	)
+	!fastcgi? ( !apache2? ( virtual/httpd-cgi ) )
+	gd? (
+		dev-perl/GD
+		dev-perl/GDTextUtil
+		dev-perl/GDGraph
+	)
+	ldap? ( dev-perl/perl-ldap )
+	mysql? ( dev-perl/DBD-mysql )
+	postgres? ( dev-perl/DBD-Pg )
+	pdf? (
+		>=dev-perl/PDF-API2-0.73
+		virtual/perl-Compress-Raw-Zlib
+	)
+	soap? (
+		dev-perl/SOAP-Lite
+		!=dev-perl/SOAP-Lite-0.711
+		!=dev-perl/SOAP-Lite-0.712
+	)
+	"
+
+OTRS_HOME="/var/lib/otrs"
+
+pkg_setup() {
+	# The enewuser otrs will fail if apache isn't there, but it's an optional dep
+	# so we create the apache user here just in case
+	enewgroup apache 81
+	enewuser apache 81 -1 /var/www apache
+	enewuser otrs -1 -1 "${OTRS_HOME}" apache
+}
+
+src_prepare() {
+	rm -r "${S}/scripts"/auto_* || die
+
+	pushd Kernel >/dev/null || die
+	local i
+	for i in *.dist; do
+		cp "${i}" $(basename "${i}" .dist) || die
+	done
+	popd >/dev/null || die
+
+	# Fix broken png file (and see pngfix help for exit codes)
+	pngfix -q --out=out.png "${S}/var/httpd/htdocs/skins/Agent/default/img/otrs-verify.png"
```

Comment:
Does upstream continue including a broken file? Was this reported upstream?

Reply:
Back then, yes. I am not sure about the upstream status and assume I could check the file we get from upstream per pngfix myself, then decide if that check/fix is still needed in the ebuild.

Reply:
I think we can drop that pngfix-block:

```
$ pngfix -w -e otrs-6.0.6/var/httpd/htdocs/skins/Agent/default/img/otrs-verify.png 
iCCP TFB default 10 11 708 1328 otrs-6.0.6/var/httpd/htdocs/skins/Agent/default/img/otrs-verify.png
IDAT OK  default 15 15 9585 36045 otrs-6.0.6/var/httpd/htdocs/skins/Agent/default/img/otrs-verify.png
```